src/forward_link.py

```diff
@@ -0,0 +1,105 @@
+from typing import List
+
+import numpy as np
+import pandas as pd
+
+
+def zerofy_values(
+    df: pd.DataFrame, target_variable: List[str] or str, expr: str
+) -> pd.DataFrame:
+    """Convert values in a dataframe column to 0 based on a python expression
+
+    Parameters
+    ----------
+    df : pd.Dataframe
+        Pandas dataframe of original data.
+    target_variable : List[str] or str
+        Column name(s) containing target variable(s).
+    query : str
+        The expression to evaluate, see here:
+        https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.eval.html
+
+    Returns
+    -------
+    df : pd.Dataframe
+
+
+    """
+
+    try:
+        df.loc[~(df.eval(expr)), target_variable] = 0
+
+    except ValueError:
+        print(
+            f"""{expr} is not a valid expression,
+        the code uses ~(df.eval({expr}) to mask the dataframe, please see here:
+        https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.eval.html
+        """
+        )
+
+
+def get_link(
```

Comment:
Again another well defined, easy to understand function with excellent doc strings.

Function name could be more specific `calculate_imputation_link`